src/audiences/edit/index.js

```diff
@@ -0,0 +1,235 @@
+										<Fragment>
+											{ [ '=', '!=' ].indexOf( rule.operator ) >= 0 && (
+												<select
+													className="audience-editor__rule-value"
+													onChange={ e => updateRule( groupId, ruleId, { value: e.target.value } ) }
+													value={ rule.value }
+													name={`audience[groups][${groupId}][rules][${ruleId}][value]`}
+												>
+													<option value="">{ __( 'Empty', 'altis-analytics' ) }</option>
+													{ fields.filter( field => field.name === rule.field ).map( field => field.data && field.data.map( ( datum, index ) => (
+														<option key={ index } value={ datum.value }>{ datum.value }</option>
+													) ) ) }
+												</select>
+											) }
+											{ [ '*=', '!*', '^=' ].indexOf( rule.operator ) >= 0 && (
+												<input
+													className="regular-text"
+													type="text"
+													value={ rule.value }
+													onChange={ e => updateRule( groupId, ruleId, { value: e.target.value } ) }
+													name={`audience[groups][${groupId}][rules][${ruleId}][value]`}
+												/>
+											) }
+										</Fragment>
+									) }
+									{ currentField.type === 'number' && (
+										<input
+											className="regular-text"
+											type="number"
+											placeholder={ `${ __( 'Average value: ', 'altis-analytics' ) } ${ currentField.stats.avg }` }
+											value={ rule.value }
+											onChange={ e => updateRule( groupId, ruleId, { value: e.target.value } ) }
+											name={`audience[groups][${groupId}][rules][${ruleId}][value]`}
+										/>
+									) }
+
+									{ group.rules.length > 1 && (
+										<Button
+											className="audience-editor__rule-remove"
+											isDestructive={ true }
+											isLink={ true }
+											onClick={ () => {
+												const newRules = group.rules.slice();
+												newRules.splice( ruleId, 1 );
+												updateGroup( groupId, { rules: newRules } );
+											} }
+										>
+											<span className="screen-reader-text">{ __( 'Remove rule', 'altis-analytics' ) }</span>
+											&times;
+										</Button>
+									) }
+								</Rule>
+							);
+						} ) }
+
+						<Button
+							className="audience-editor__rule-add"
+							isLarge={ true }
+							onClick={ () => updateGroup( groupId, { rules: group.rules.concat( [ defaultRule ] ) } ) }
+						>
+							{ __( 'Add a rule', 'altis-analytics' ) }
+						</Button>
+
+						{ audience.groups.length > 1 && (
+							<Button
+								className="audience-editor__group-remove"
+								isDestructive={ true }
+								isLink={ true }
+								onClick={ () => {
+									const newGroups = audience.groups.slice();
+									newGroups.splice( groupId, 1 );
+									updateAudience( { groups: newGroups } );
+								} }
+							>
+								{ __( 'Remove group', 'altis-analytics' ) }
+							</Button>
+						) }
+					</Group>
+				);
+			} ) }
```

Comment:
This whole section is massive and could use refactoring into reusable components instead to enhance readability.